Sources/MongoSwift/BSON/DocumentIterator.swift

```diff
@@ -190,6 +212,35 @@ public class DocumentIterator: IteratorProtocol {
 #endif
     }
 
+    /// Internal helper function for copying elements from some source 'bson_t' to a destination 'bson_t' while
+    /// excluding a non-zero number of keys
+    internal static func copyElements(
```

Comment:
It might be a little more Swift-like to write this as an instance method on `Document` that takes in the source `Document`  as a parameter.
e.g.
```swift
let doc = Document()
doc.copyElements(from: otherDoc, excluding: ["key"])
```
To access the underlying pointers, you can use the free-floating [`withMutableBSONPointer`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/BSON/Document.swift#L512) function for mutable access or the [`_bson`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/BSON/Document.swift#L53) property on `Document` for read-only access. This makes me realize we should probably move both of these onto `Document` and use similar names for both of them.

In general, we try to limit working directly with libbson / libmongoc wherever possible and instead use the abstractions we've written around them. When we do need to drop into C, we try to do it within `withXPointer` blocks. An example of this is in [`ListDatabasesOperation`](https://github.com/mongodb/mongo-swift-driver/blob/master/Sources/MongoSwift/Operations/ListDatabasesOperation.swift#L59).

Reply:
Done.